### DISCUSSION TITLE
能快速拿到单元格编辑的前后内容方法吗

### DISCUSSION THREAD

**Discussion by fddqfddq** [dream-num/univer#2261] [Q&A | 常见问题]
就是要个更新功能，如果单元格有变更就记录下来，当前我看文档有以下3个内容：
单元格进入编辑，没有单元格信息
univerAPI.onCommandExecuted((command) => {
    if(command.id === 'sheet.operation.set-cell-edit-visible' && command.params.visible){
        console.log('Cell edit visible')
    }
})
单元格退出编辑，也没有单元格信息
univerAPI.onCommandExecuted((command) => {
    if(command.id === 'sheet.operation.set-cell-edit-visible' && !command.params.visible){
        console.log('Cell edit invisible')
    }
})
监听选区变化，这个selection里有r,c，但是拿不到当前单元格的值吗？
const activeWorkbook = univerAPI.getActiveWorkbook();
activeWorkbook.onSelectionChange((selection) => {
  console.log(selection);
});

**Comment by fddqfddq** [comment]
有类似Luckysheet中的这个吗？
   //作用：更新这个单元格后触发参数：
         cellUpdated: function (r, c, oldValue, newValue, isRefresh){}

**Comment by JackieCheung** [comment]
同求

**Comment by fddqfddq** [comment]
算了，时间紧张，暂时没找到好的办法，临时解决办法是
初始化时记录原始数据
activeWorkbook.onSelectionChange时记录单元格
save时再结合前面2点，对记录单元格的值前后变化作比较

**Comment by jikkai** [comment]
https://stackblitz.com/edit/github-aayzou-qbip5w

**Comment by fddqfddq** [comment]
`if (id === 'doc.command.insert-text' || id === 'doc.command.delete-text') {
        const doc = univerAPI.getActiveDocument();
        if (doc) {
          const snapshot = doc.getSnapshot();
          const activeRange = univerAPI.getActiveWorkbook().getActiveSheet().getSelection().getActiveRange();
          const sheetname = univerAPI.getActiveWorkbook().getActiveSheet().getSheetName();
          const r = activeRange.getRow();
          const c = activeRange.getColumn();
          const v = snapshot.body?.dataStream.replace(/\s+/g, '');;
         }
}`
